Apps/JobScheduler/src/Jobs/CommunicationJob.cs

```diff
@@ -0,0 +1,155 @@
+        {
+            this.logger.LogDebug($"Creating emails & communication emails...");
+            List<Communication> communications = this.communicationDelegate.GetEmailCommunicationsToSend();
+
+            if (communications.Count > 0)
+            {
+                this.logger.LogInformation($"Found {communications.Count} communications which need to be processed / re-processed.");
+
+                foreach (Communication communication in communications)
+                {
+#pragma warning disable CA1031 //We want to catch exception.
+                    try
+                    {
+                        List<UserProfile> usersToSendCommEmails = new List<UserProfile>();
+                        bool moreUsersToCreateCommunicationEmails = false;
+                        do
+                        {
+                            usersToSendCommEmails = this.commEmailDelegate.GetActiveUserProfilesByCommunicationId(communication.Id, this.maxFetchSize);
+                            foreach (UserProfile profile in usersToSendCommEmails)
+                            {
+                                // Insert a new Email record into db.
+                                Email email = new Email()
+                                {
+                                    From = this.fromEmailAddressHGDonotreply,
+                                    To = profile.Email,
+                                    Subject = communication.Subject,
+                                    Body = communication.Text,
+                                    FormatCode = EmailFormat.HTML,
+                                    Priority = communication.Priority,
+                                };
+                                this.emailQueueService.QueueNewEmail(email, false);
+
+                                // Inser a new CommunicationEmail record into
+                                CommunicationEmail commEmail = new CommunicationEmail()
+                                {
+                                    Email = email,
+                                    UserProfile = profile,
+                                    Communication = communication,
+                                };
+                                this.commEmailDelegate.Add(commEmail, false);
+                            }
+
+                            if (communication.CommunicationStatusCode != CommunicationStatus.Processing)
+                            {
+                                communication.CommunicationStatusCode = CommunicationStatus.Processing;
+                                this.communicationDelegate.Update(communication, false);
+                            }
+
+                            this.dbContext.SaveChanges(); // commit after every retryFetchSize (or 250) pairs of Email & CommunicationEmail.
+
+                            moreUsersToCreateCommunicationEmails = usersToSendCommEmails.Count == this.maxFetchSize ? true : false;
```

Comment:
The ? Operator is unnecessary

Reply:
you are right. I 'm deleting " ? true : false;".